### ISSUE TITLE
Release 2024.10.1 uses much more CPU

### ISSUE THREAD

**Issue by j64bit** [goauthentik/authentik#11965]
Authentik is a great tool and I'd to start with thanking all involved for making it available to home users like myself!

**Describe the bug**
After upgrading to "Release 2024.10.1" CPU usage is dramatically increased already idle state.

![grafik](https://github.com/user-attachments/assets/a4fe0a42-2a8c-49df-b23b-478bce2dbc56)

I use the docker image in an alpine linux lxc with promox on a intel laptop.

**To Reproduce**
Install the version and observe CPU usage

**Expected behavior**
Unchanged CPU usage

**Screenshots**
See above 

**Logs**
none

**Version and Deployment (please complete the following information):**

-   authentik version: 2024.10.1 
-   Deployment: docker-compose in proxmox environment 
 
**Additional context**
I reverted back to previous version.

**Comment by EHRETic** [comment]
Hi,

I can confirm that behaviour too, climbed right after the update on my Docker VM:

![image](https://github.com/user-attachments/assets/0cd58e4a-bee0-4328-95ea-96f18cb541ad)

Don't if it helps, but this is the process taking CPU on my VM:

![image](https://github.com/user-attachments/assets/9f3ae4fc-b9f5-481d-bf40-cb3332c51e9c)

Seems to come from worker:

![image](https://github.com/user-attachments/assets/48114b70-5e0d-4efd-99f4-62db6209256f)

Found somethin about Celery in logs:

```
2024-11-08T09:10:38.361545149Z warning | event=/ak-root/venv/lib/python3.12/site-packages/celery/worker/consumer/consumer.py:508: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
whether broker connection retries are made during startup in Celery 6.0 and above.
If you wish to retain the existing behavior for retrying connections on startup,
you should set broker_connection_retry_on_startup to True.
  warnings.warn(
 logger=py.warnings timestamp=1731057038.3590796 
2024-11-08T09:10:38.378573216Z warning | event=/ak-root/venv/lib/python3.12/site-packages/celery/worker/consumer/consumer.py:508: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
whether broker connection retries are made during startup in Celery 6.0 and above.
If you wish to retain the existing behavior for retrying connections on startup,
you should set broker_connection_retry_on_startup to True.
  warnings.warn(
 logger=py.warnings timestamp=1731057038.3783574 
```

Let me know if you need more (but I might help to get more!) 😊

**Comment by j64bit** [comment]
Interesting. Did you try setting "broker_connection_retry_on_startup" to False (to avoid the retries - if get this right)?

**Comment by EHRETic** [comment]
> Interesting. Did you try setting "broker_connection_retry_on_startup" to False (to avoid the retries - if I got this right)?

Hi, I just tried but it didn't change anything (tried the setting in both server & worker containers)

What can I provide you now?

**Comment by j64bit** [comment]
ah, ok. thanks.
i just noticed i am still using postgress 12.  in past months there was a change to postgress 16. 
Maybe that is the source. I will try to do the upgrade now ...
As per this:  https://docs.goauthentik.io/docs/troubleshooting/postgres/upgrade_docker

**Comment by EHRETic** [comment]
> ah, ok. thanks. i just noticed i am still using postgress 12. in past months there was a change to postgress 16. 

I'm using PG16 since the start on both of my instances, so I don't think it might change something.

**Comment by j64bit** [comment]
correct - no improvement after upgrade to 16 for postgres.
Looks like workers are being started continiuosly one by one - as in your screenshots.

**Comment by EHRETic** [comment]
Strangely, rolling back to 2024.8 doesn't solve the CPU issue in my test instance (rolling back from updated instance)
Restore would work.

**Comment by j64bit** [comment]
sorry wrong button was hit

**Comment by j64bit** [comment]
sorry, i hit the wrong button

**Comment by EHRETic** [comment]
I'll try to test it tomorrow too.

What I noticed is that my Docker VM, after Authentik restart will have 20 minutes high CPU because of celery process and then will be alright. 
Don't know why.

**Comment by EHRETic** [comment]
> I had the same problem and I was actually able to "fix" it limiting the ulimits like described here: [celery/celery#8306 (comment)](https://github.com/celery/celery/issues/8306#issuecomment-1744722222)
> 
> Can someone confirm this?

Did you add these ulimit values in Authentik Worker container settings? Or somewhere else?
(Just asking before being able to try it out)

**Comment by chrisliebaer** [comment]
To provide a bit more details, it appears that celery/billiard is **brute forcing file handles**. And the more file handles it has, the longer it takes for it to be done. There is also a related issue referencing the issue I linked if you are interested. I don't even

**Comment by j64bit** [comment]
Tried it - with postgres still in version 12

Unfortunately it didnt work for me. Still have on the worker:

`Load average: 1.29 1.15 0.91 3/811 1735622`
`PID  PPID USER     STAT   VSZ %VSZ CPU %CPU COMMAND`
`556881 552401 1000   S     803m  39%   1  30% [celeryd: celery@a379c6a12d8a:ForkPoolWorker-33]`
`552401 552312 1000   S     582m  28%   1   4% [celeryd: celery@a379c6a12d8a:MainProcess] -active- ()`
`552124 552087 999    S    63872   3%   1   1% redis-server *:6379`

**Comment by EHRETic** [comment]
Hi there,

I did the test... red, without, green with the settings... I think it works "a bit" better! 😉
(without, CPU will come down after 20-30 minutes)

![image](https://github.com/user-attachments/assets/097cdfb1-8b81-4572-bf47-6856ae07aae4)

**Comment by BeryJu** [comment]
most likely related to #7762, see https://github.com/goauthentik/authentik/pull/7762#issuecomment-2143361434 for why that hasn't been merged

**Comment by EHRETic** [comment]
> most likely related to #7762, see [#7762 (comment)](https://github.com/goauthentik/authentik/pull/7762#issuecomment-2143361434) for why that hasn't been merged

I agree, it's not a real fix, but is there here something we can/should address to Celery directly?
Waaayyy out of my (Authentik user) league! 😜
